### DISCUSSION TITLE
Binary of LSAN version failed to run

### DISCUSSION THREAD

**Discussion by xuwei0912** [apache/doris#31284] [A - General / Q&A]
版本2.0.4
报错信息：
CMake Error at src/clucene/src/shared/cmake/CheckSnprintf.cmake:20 (MESSAGE):
  snprintf has a bug, and we don't have a bug, and we don't have a replacement!
Call Stack (most recent call first):
  src/clucene/src/shared/CMakeLists.txt:249 (CHECK_SNPRINTF)

**Comment by LemonLiTree** [comment]
检查下依赖库的版本

**Reply by xuwei0912** [reply]
相同的环境可以成功编译debug版本

**Comment by Userwhite** [comment]
+1 想跑ut跑不了

**Comment by mychaow** [comment]
修改clucene的 src/shared/CMakeLists.txt 文件，禁掉leak check
```
set(ASAN_OPTIONS_ORIGINAL "$ENV{ASAN_OPTIONS}")
set(ENV{ASAN_OPTIONS} "detect_leaks=0")
...
#check if snprintf functions are buggy
CHECK_SNPRINTF()

set(ENV{ASAN_OPTIONS} "${ASAN_OPTIONS_ORIGINAL}")
```

**Reply by xuwei0912** [reply]
也还是报错呢

**Reply by mychaow** [reply]
确定改的是clucene的CMakeFile.txt么？报错还是检查snprintf?，添加禁止leak check操作需要在check_snprintf()之前

**Reply by xuwei0912** [reply]
嗯，这样改的：
（clucene/src/shared/CMakeLists.txt） 
set(ASAN_OPTIONS_ORIGINAL "$ENV{ASAN_OPTIONS}")
set(ENV{ASAN_OPTIONS} "detect_leaks=0")
**#check if snprintf functions are buggy
CHECK_SNPRINTF()**
set(ENV{ASAN_OPTIONS} "${ASAN_OPTIONS_ORIGINAL}")

**Reply by xuwei0912** [reply]
报错还是检查snprintf

**Comment by Vallishp** [comment]
i had similar issue. i disable asan for clucene and continued. 
Commented ASAN and LSAN settings and hardcoded Debug for clucene.

doris/be/src/clucene/CMakeLists.txt

![image](https://github.com/apache/doris/assets/11808387/3b4398ef-815c-43a6-a6fa-709462c3c549)
